config/v1/types_feature.go

```diff
@@ -125,18 +125,61 @@ type FeatureGateDetails struct {
 	// disabled is a list of all feature gates that are disabled in the cluster for the named version.
 	// +optional
 	Disabled []FeatureGateAttributes `json:"disabled"`
+	// renderedMinimumComponentVersions are the component versions that the feature gate list of this status were rendered from.
+	// Currently, the only supported component is Kubelet, and setting a required minimum kubelet component will set the
+	// minimumKubeletVersion field in the nodes.config.openshift.io CRD.
+	// +kubebuilder:validation:MaxItems:=1
+	// +listType=map
+	// +listMapKey=component
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	// +optional
+	RenderedMinimumComponentVersions []MinimumComponentVersion `json:"renderedMinimumComponentVersions,omitempty"`
 }
 
 type FeatureGateAttributes struct {
 	// name is the name of the FeatureGate.
 	// +required
 	Name FeatureGateName `json:"name"`
 
+	// requiredMinimumComponentVersions is a list of component/version pairs that declares the is the lowest version the given
+	// component may be in this cluster.
+	// Currently, the only supported component is Kubelet, and setting a required minimum kubelet component will set the
+	// minimumKubeletVersion field in the nodes.config.openshift.io CRD.
+	// +kubebuilder:validation:MaxItems:=1
+	// +listType=map
+	// +listMapKey=component
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	// +optional
+	RequiredMinimumComponentVersions []MinimumComponentVersion `json:"requiredMinimumComponentVersions,omitempty"`
+
 	// possible (probable?) future additions include
 	// 1. support level (Stable, ServiceDeliveryOnly, TechPreview, DevPreview)
 	// 2. description
 }
 
+// MinimumComponentVersion is a pair of Component and Version that specifies the required minimum Version of the given Component
+// to enable this feature.
+type MinimumComponentVersion struct {
+	// component is the entity whose version must be above a certain version.
+	// +required
+	Component MinimumComponent `json:"component"`
+	// version is the minimum version the given component may be in this cluster.
+	// version must be in semver format (x.y.z) and must consist only of numbers and periods (.).
```

Comment:
Is there a way to explain that this version is the Kubernetes version, so we know it's going to be ~1.32 and not ~4.19?

Reply:
how's that look?

Reply:
That's great, thanks